.github/workflows/CI.yml

```diff
@@ -186,3 +186,50 @@ jobs:
           name: prebuilds
           pattern: prebuilds-*
           delete-merged: true
+  smoke-test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+          - macos-latest
+        pm:
+          - npm
+          - yarn
+          - pnpm
+    steps:
+      - uses: actions/setup-node@v4
```

Comment:
This can include multiple versions such as `10, 12, 18, 22` to ensure compatibility across different versions.

Reply:
You're absolutely right and I'd like to get there before this is merged. As you can see from my embarrassing commit history, I'm not quite there yet...